### ISSUE TITLE
add functionality to detect introduction of NAN in transform method of discretizers

### ISSUE THREAD

**Issue by solegalli** [feature-engine/feature_engine#295]
For the arbitrary, equal width and equal frequency, we should include in the transform method a check to see if NaN are being introduced accidentally.

This can happen if a value is outside the boundaries entered by the user in the arbitrary, or of the variable is too skewed for the other transformers.

The catch should be as a warning to begin with, not to break backwards compatibility perhaps, and should inform the user exactly in which variables the NaN were introduced.

We should also expand the tests of all the transformers to make sure this functionality works. As it is the same functionality for all transformers, and also for the categorical encoders as well as per issue #294, I wonder if we should make a master function from this? I think it depends in the amount of code, if it is a one liner probably not.

**Comment by solegalli** [comment]
At the back of #169

**Comment by solegalli** [comment]
Hi @Morgan-Sell

Would you like to pick up this issue? 

It is very similar to the one we just worked on. 

We need to expand the error message to clarify which variables are the ones where nan were introduced.

**Comment by Morgan-Sell** [comment]
Por supuesto!

Confirming the high-level scope:
- Create a new branch, PR, etc.
- Assess whether there is a NaN value in the dataframe that is inputted in the `transform()` methods for both equal_frequency.py and equal_width.py.
- Create a warning that is in the same vein as the one we developed for the "base_enconder NaN detection" issue.
- Implement the appropriate tests using `pytest.raises` in both test_equal_frequency_discretiser.py and test_equal_width_frequency.py.

**Comment by Morgan-Sell** [comment]
Ah, yes, class inheritance! Thanks for the framework!

**Comment by Morgan-Sell** [comment]
@solegalli, I thought I created a PR for this task. I went to the branch called "discretizer-nan-detection", which I created for this task, and selected "Open pull request".

<img width="927" alt="Screen Shot 2021-12-16 at 4 38 21 PM" src="https://user-images.githubusercontent.com/5661386/146469595-c6d536c9-b7c6-468e-b2c6-f67ae27c0c10.png">

However, I do not see a PR. I've already committed two edits to the "discretizer-nan-detection" branch.

What am I missing? Thanks!
